modules/lvgl/lvgl.c

```diff
@@ -222,28 +246,26 @@ static int lvgl_init(void)
 	disp_data.display_dev = display_dev;
 	display_get_capabilities(display_dev, &disp_data.cap);
 
-	lv_disp_drv_init(&disp_drv);
-	disp_drv.user_data = (void *)&disp_data;
-
-#ifdef CONFIG_LV_Z_FULL_REFRESH
-	disp_drv.full_refresh = 1;
-#endif
-
-	err = lvgl_allocate_rendering_buffers(&disp_drv);
-	if (err != 0) {
-		return err;
+	display = lv_display_create(disp_data.cap.x_resolution, disp_data.cap.y_resolution);
+	if (!display) {
```

Comment:
I think failure on the display creation could be reported as -ENOMEM or -ENODEV since -ENOTSUP tends to indicate the actual function is not supported, which seems not the case.

Reply:
The error code is better true 👍 , fixed.